File: clients/php/src/ZawgyiDetector.php
<?php
/**
 * Created by SteveNay.
 * User: Nay Lin Aung
 * Date: 5/26/2018
 * Time: 4:10 PM
 */

namespace Googlei18n\MyanmarTools;

use Exception;

final class ZawgyiDetector
{
    private $model = 0;

    public function __construct()
    {
        try {
            $fileName = join(DIRECTORY_SEPARATOR, [__DIR__, "..", "resources", "zawgyiUnicodeModel.dat"]);
            // file open with read permission with binary format
            $inStream = fopen($fileName, "rb");

            if ($inStream == null) {
                // IO Exception
                throw new Exception("Model file zawgyiUnicodeModel.dat not found");
            }

            $this->model = new ZawgyiUnicodeMarkovModel($inStream);
            fclose($inStream);

        } catch (Exception $exception) {
            throw new Exception("Could not load Markov model from resource file", 0, $exception);
        }
    }

    /**
     * Performs detection on the given string. Returns the probability that the string is Zawgyi given
     * that it is either Unicode or Zawgyi. Values approaching 1 are strong Zawgyi; values approaching
     * 0 are strong Unicode; and values close to 0.5 are toss-ups.
     *
     * <p>If the string does not contain any Myanmar range code points, -Infinity is returned.
     *
     * @param string $input The string on which to run detection.
     * @param bool $verbose If true, print debugging information to standard output.
     * @return float The probability that the string is Zawgyi (between 0 and 1), or -Infinity if the string
     *     contains no Myanmar range code points.
     */
    public function getZawgyiProbability(string $input, bool $verbose = false): float
    {
        return $this->model->predict($input, $verbose);
    }
}

File: clients/dart/lib/src/zawgyi_detector.dart
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';
import 'markov/zawgyi_unicode_markov_model.dart';

class ZawGyiDetector{
  final ZawGyiUnicodeMarkovModel model;

  ZawGyiDetector._(this.model);

  static Future<ZawGyiDetector> create({String package='myanmar_tools'}) async {
    try {
      var data = await rootBundle.load('packages/$package/resources/zawgyiUnicodeModel.dat',);
      return ZawGyiDetector._(ZawGyiUnicodeMarkovModel(data));
    }catch(e,stacktrace){
      throw FlutterError('Could not load Markov model from resource file . \n$e\n $stacktrace');
    }
  }

  @protected
  @visibleForTesting
  //Only for Testing
  static Future<ZawGyiDetector> createForTest() async {
    try {
      var data = await rootBundle.load('resources/zawgyiUnicodeModel.dat',);
      return ZawGyiDetector._(ZawGyiUnicodeMarkovModel(data));
    }catch(e,stacktrace){
      throw FlutterError('Could not load Markov model from resource file . \n$e\n $stacktrace');
    }
  }

  /// Performs detection on the given string. Returns the probability that the string is Zawgyi given
  /// that it is either Unicode or Zawgyi. Values approaching 1 are strong Zawgyi; values approaching
  /// 0 are strong Unicode; and values close to 0.5 are toss-ups.
  ///
  /// If the string does not contain any Myanmar range code points, double.negativeInfinity is returned.
  ///
  /// * @param input The string on which to run detection.
  /// * @return The probability that the string is Zawgyi (between 0 and 1), or -Infinity if the string contains no Myanmar range code points.
  /// <p>ပေးထားသော စာသားအား ယူနီကုတ် သို့မဟုတ် ဇော်ဂျီ ဖြစ်ကြောင်းစစ်ဆေးရန်။
  /// <p>ဇော်ဂျီ စာသား ဖြစ်ပါက ၁ နှင့် အနီးပတ်ဝန်းကျင် ဂဏန်း ကိုရရှိပြီး ၊ ယူနီကုတ် စာသား ဖြစ်ပါက ၀ နှင့် အနီးပတ်ဝန်းကျင် ဂဏန်း ကိုရရှိမည် ဖြစ်ပါသည်။
  /// <p>၀.၅ အနီးပတ်ဝန်းကျင် ဂဏန်း ဖြစ်ပါက ယူနီကုတ်/ဇော်ဂျီ  နှစ်ခုစလုံး ဖြစ်နိုင်ပြီး လိုအပ်သလိုချိန်ဆ သုံးသွားနိုင်ပါသည်။
  ///  <p>ပေးထားသော စာသားတွင် ယူနီကုတ် နှင့် ဇော်ဂျီ  နှစ်ခုစလုံး မပါရှိပါက double.negativeInfinity ကိုရရှိမည် ဖြစ်ပါသည်။
  /// * @param input ယူနီကုတ် သို့မဟုတ် ဇော်ဂျီ ဖြစ်ကြောင်းစစ်ဆေးရန် စာသား
  /// * @return ဇော်ဂျီ စာသား ဖြစ်နိုင်ခြေ (၀ နှင့် ၁ ကြား) ၊ သို့မဟုတ် ယူနီကုတ် နှင့် ဇော်ဂျီ  နှစ်ခုစလုံး မပါရှိပါက double.negativeInfinity
  double getZawGyiProbability(String input) {
  return model.predict(input);
  }
}